### PR TITLE
CIV - 5237

### PR DESCRIPTION
**Pull request by suvarnathekkoot**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-5237

### Change description ###

PBA payment

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by drummondjm — APPROVED** [review]

File: config/owasp/suppressions.xml
Comment (filippo-possenti-version1) [inline config/owasp/suppressions.xml:96]: Okay, but open a ticket for these suppressions and assign it to me, so I can look into them.

File: src/main/java/uk/gov/hmcts/reform/civil/model/hearing/HFPbaDetails.java
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/model/hearing/HFPbaDetails.java:41]: I believe this can be simplified.
Take a look at the classes annotated with `@Jacksonized`.

Is there a particular reason why you chose this approach instead of just putting the jackson annotations on the individual properties and do away with the dedicated constructor?

File: src/main/java/uk/gov/hmcts/reform/civil/service/PaymentsService.java
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/service/PaymentsService.java:80]: The content of the `if` and the `else if` is completely identical.

If it's meant to be, you don't need the `if` and `else if`. If it's not meant to be, then you have a problem.

Please take a look and either remove the unnecessary code or fix the issue.

Comment (filippo-possenti-version1) [reply src/main/java/uk/gov/hmcts/reform/civil/service/PaymentsService.java:80]: Also: install the SonarLint plugin on your IntelliJ, as it will detect these situations automatically for you.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandlerTest.java
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandlerTest.java:36]: Please add a test for the `camundaActivityId` method as well. I know it seems trivial, but it can intercept typos or people modifying values they shouldn't be touching.

Also: the `catch` clause of `makePaymentServiceReq` is not intercepted by your tests, meaning you are only testing the happy path. Please add a test that simulates a `FeignException` to verify that everything is correct also in that case.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/HearingScheduledHandlerTest.java
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/HearingScheduledHandlerTest.java:167]: Only white-space changes here. If you rollback this file, it's one less to be reviewed.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandlerTest.java
Comment (filippo-possenti-version1) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandlerTest.java:36]: A note: using IntelliJ's feature to run tests with coverage will help you detect these situations. After running the tests in that mode, IntelliJ will highlight in green, yellow or red the lines that have been hit during your tests. This will guide you to getting to 100% coverage.

File: src/test/java/uk/gov/hmcts/reform/civil/service/PaymentsServiceTest.java
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/service/PaymentsServiceTest.java:265]: This unit test failed with a NullPointerException to me. Please take a look.

Here's a portion of the stack trace for your convenience:
```
java.lang.NullPointerException
	at uk.gov.hmcts.reform.civil.service.PaymentsService.createCreditAccountPayment(PaymentsService.java:85)
	at uk.gov.hmcts.reform.civil.service.PaymentsServiceTest$Specified.shouldCreateCreditAccountPayment_whenValidCaseDetails(PaymentsServiceTest.java:156)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        [...]
```

**Review by filippo-possenti-version1 — CHANGES_REQUESTED** [review]
Some simplification, dead code and unit test failures need to be addressed here before I can give approval.

Feel free to give me a call if you need any help or guidance on this.

**Pull request closed without merging**
